handlers/handlers.go

```diff
@@ -17,25 +18,28 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
-type StreamInfo struct {
-	callbackUrl string
-}
-
 type CatalystAPIHandlersCollection struct {
 	MistClient  MistAPIClient
-	StreamCache map[string]StreamInfo
+	StreamCache *StreamCache
 }
 
 func (d *CatalystAPIHandlersCollection) Ok() httprouter.Handle {
-	return func(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	return func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
 		io.WriteString(w, "OK")
 	}
 }
 
-func (d *CatalystAPIHandlersCollection) TranscodeSegment() httprouter.Handle {
+// Takes mpegts segment as input, ingests in Mist under SOURCE_PREFIX streamName, transcodes into RENDITION_PREFIX stream then pushes renditions to specified destination.
+// Manually starts MistLivepeerProc binary. For this we require port of B-node that listens on 127.0.0.1 .
+// When `MistProcLivepeer` starts, SOURCE_PREFIX stream is taken as input and (will automatically - WIP) start SOURCE_PREFIX stream.
```

Comment:
actually, workaround. WIP is on Mist side.

Reply:
Ok, I see. So, from this code's perspective it's not WIP, it's more like a workaround. To avoid confusion, I suggest to change it to something like:

```
// TODO: Remove `MistProcLivepeer` after this issue is fixed <link-to-related-mist-gh-isssue>
// `MistProcLivepeer` is a workaround until the automatic stream start is implemented in Mist 
```